File: 1_TwoSum/1.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hashTable = Hash.new
    
    nums.each_with_index do |value, index|
        find_value = target - value
        
        if hashTable.key?(find_value)
            return [hashTable[find_value], index]
        end
        
        hashTable[value] = index
    end
    []
end

File: 1_TwoSum/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var hashTable = new Map();

    for (let i = 0; i < nums.length; i++) {
        var x = nums[i];
        var get_value = hashTable.get(target - x);

        if (get_value !== undefined) {
            return [get_value, i];
        }

        hashTable.set(x, i);
    }

    return []
};